### PR TITLE
scan estimation orchestrator

### PR DESCRIPTION
**Pull request by fishkerez**
## Description
Resolve https://github.com/openclarity/vmclarity/issues/586

Discussion: https://github.com/openclarity/vmclarity/discussions/465

Continuing the apiserver PR: https://github.com/openclarity/vmclarity/pull/559

Adds 2 new reconcile controllers for ScanEstimation and AssetScanEstimation.

The cost calculation implementation for AWS will be added soon.

## Type of Change

[ ] Bug Fix  
[* ] New Feature  
[ ] Breaking Change  
[ ] Refactor  
[ ] Documentation  
[ ] Other (please describe)

### CODE REVIEW

**Comment by Tehsmash** [comment]
@fishkerez if you rebase this PR onto the API server one and leave this in draft there is no need for a feature branch, and we can review just the top commits for each part of the commit chain. That will make all the CI work and the top of the chain testable, and allow us to merge things directly to main without having to have a second pull request to merge it all in later.

File: pkg/orchestrator/scanestimationwatcher/watcher.go [hunk 1]
```diff
@@ -0,0 +1,566 @@
...
+		return nil, err
+	}
+
+	assetWriteable := &models.Asset{
+		Id:         asset.Id,
+		Revision:   asset.Revision,
+		ScansCount: asset.ScansCount,
+		Summary:    asset.Summary,
+		AssetInfo:  asset.AssetInfo,
+	}
```
Comment (Tehsmash) [inline pkg/orchestrator/scanestimationwatcher/watcher.go:345]: This should be a relationship to an asset in the system, we should not be duplicating any asset information in the AssetScanEstimation.

**Review by Tehsmash — CHANGES_REQUESTED** [review]

**Comment by fishkerez** [comment]
> @fishkerez if you rebase this PR onto the API server one and leave this in draft there is no need for a feature branch, and we can review just the top commits for each part of the commit chain. That will make all the CI work and the top of the chain testable, and allow us to merge things directly to main without having to have a second pull request to merge it all in later.

I will open a new PR after I will finish the work on. the

File: pkg/orchestrator/scanestimationwatcher/watcher.go [hunk 2]
```diff
@@ -0,0 +1,566 @@
...
+
+	assetErrs := make([]error, 0, numOfAssets)
+	for err := range errs {
+		assetErrs = append(assetErrs, err)
+	}
+	numOfErrs := len(assetErrs)
+
+	if numOfErrs > 0 {
+		return fmt.Errorf("failed to create %d AssetScanEstimation(s) for ScanEstimation. ScanEstimationID=%s: %w", numOfErrs, *scanEstimation.Id, assetErrs[0])
+	}
```
[the remark this answers is not in the document]
Comment (fishkerez) [reply pkg/orchestrator/scanestimationwatcher/watcher.go:317]: Took it from scanWatcher reconcile logic.

**Comment by fishkerez** [comment]
Moved to  https://github.com/openclarity/vmclarity/pull/603

**Pull request closed without merging**